### PR TITLE
Metrics implementation for prometheus & spring

### PR DESCRIPTION
**Pull request by hendrikebbers**
Metrics implementation based on prometheus
Metrics support in Spring
Noop metrics

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/892)

### CODE REVIEW

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/module/MetricsServlet.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private final PrometheusMeterRegistry prometheusRegistry;
+
+    public MetricsServlet(final PrometheusMeterRegistry prometheusRegistry) {
+        this.prometheusRegistry = Assert.requireNonNull(prometheusRegistry, "prometheusRegistry");
+    }
+
+    @Override
+    protected void service(final HttpServletRequest req, final HttpServletResponse resp) throws ServletException, IOException {
+        LOG.debug("Metrics servlet called");
+        String response = prometheusRegistry.scrape();
```
Comment (giftkugel) [inline platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/module/MetricsServlet.java]: Add `final` keyword.

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/bootstrap/SimpleConfigurationProvider.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package com.canoo.dp.impl.server.bootstrap;
+
+import com.canoo.platform.server.spi.ConfigurationProvider;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SimpleConfigurationProvider implements ConfigurationProvider {
```
Comment (giftkugel) [inline platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/bootstrap/SimpleConfigurationProvider.java]: Add `final` keywords to the method declarations

File: platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/module/MetricsServlet.java (hunk 1 above)
Comment (hendrikebbers) [reply platform-extras/dolphin-platform-metrics-server/src/main/java/com/canoo/dp/impl/platform/server/metrics/module/MetricsServlet.java]: done

File: platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/bootstrap/SimpleConfigurationProvider.java (hunk 2 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-server/src/main/java/com/canoo/dp/impl/server/bootstrap/SimpleConfigurationProvider.java]: done

**Pull request merged**